packages/fuels-core/src/code_gen/custom_types_gen.rs

```diff
@@ -192,8 +192,34 @@ pub fn expand_custom_enum(name: &str, prop: &Property) -> Result<TokenStream, Er
         match param_type {
             // Case where an enum takes another enum
             ParamType::Enum(_params) => {
```

Comment:
Unless I am mistaken, I don't think this supports deeply nested enums, but rather just depth=2. Can you please provide tests to show that it's supported, or implement the support for "arbitrarily" (n=3 is fine for tests) deep nested enums.

Reply:
I have added the test.